### PR TITLE
Make it build on linux + various fixes

### PR DESCRIPTION
**Pull request by iacore**
The `EX` macro is messing with the code, so I deleted one.

Everything else should be self-explanatory.

### CODE REVIEW

File: hyperpoint.cpp
Comment (Quuxplusone) [inline hyperpoint.cpp:25]: Probably obvious to @zenorogue, but just in case: obviously don't merge this change, unless you're OK with `golden_phi` becoming `0.0`. 😛

**Review by Quuxplusone — CHANGES_REQUESTED** [review]

File: hyperpoint.cpp
Comment (iacore) [reply hyperpoint.cpp:25]: oh wait

**Comment by iacore** [comment]
@zenorogue can you take a look at this?

**Comment by zenorogue** [comment]
Everything looks good. Merged, thanks!

**Pull request merged**
